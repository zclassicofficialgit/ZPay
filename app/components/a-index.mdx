---
name: Home
route: /
---

# Zepio Wallet

Use the sidebar menu to browse the component library of React components available for the Zepio application.

> This is not an exhaustive list, and we're always looking for folks to help keep it updated.
