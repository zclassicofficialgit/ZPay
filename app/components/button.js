// @flow

import React, { type ElementProps } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const DefaultButton = styled.button`
  align-items: center;
  display: flex;
  justify-content: center;
  padding: 10px 30px;
  font-family: ${props => props.theme.fontFamily};
  font-weight: ${props => String(props.theme.fontWeight.bold)};
  font-size: ${props => `${props.theme.fontSize.regular}em`};
  cursor: pointer;
  outline: none;
  min-width: 100px;
  border-radius: ${props => props.theme.boxBorderRadius};
  transition: background-color 0.1s ${props => props.theme.transitionEase};
`;

const Primary = styled(DefaultButton)`
  background-color: ${props => props.theme.colors.buttonPrimaryBg};
  color: ${props => props.theme.colors.buttonPrimaryText};
  border: none;

  &:hover {
    opacity: 0.9;
  }

  &:disabled {
    background-color: ${props => props.theme.colors.buttonPrimaryDisabledBg};
    cursor: not-allowed;
    border: none;
    opacity: 0.45;

    &:hover {
      opacity: 0.45;
    }
  }
`;

const Secondary = styled(DefaultButton)`
  background-color: ${props => props.theme.colors.buttonSecondaryBg};
  color: ${props => props.theme.colors.buttonSecondaryText};
  border: 1px solid ${props => props.theme.colors.buttonSecondaryBorder};

  &:hover {
    background-color: ${props => props.theme.colors.buttonSecondaryHoveredBg}
    border-color: ${props => props.theme.colors.buttonSecondaryHoveredBg}
  }

  &:disabled {
    background-color: ${props => props.theme.colors.buttonSecondaryDisabledBg};
    cursor: not-allowed;
    color: ${props => props.theme.colors.buttonBorderColor};

    &:hover {
      border-color: ${props => props.theme.colors.buttonBorderColor};
    }
  }
`;

const Icon = styled.img`
  height: 12px;
  width: 12px;
  margin-right: 10px;
`;

type Props = {
  ...ElementProps<'button'>,
  label: string,
  onClick?: () => void,
  to?: ?string,
  variant?: 'primary' | 'secondary',
  disabled?: boolean,
  icon?: string | null,
  className?: string,
  isLoading?: boolean,
  id?: string,
};

export const Button = ({
  onClick,
  label,
  to,
  variant,
  disabled,
  icon,
  className,
  isLoading,
  id,
}: Props) => {
  if (to && onClick) throw new Error('Should define either "to" or "onClick"');

  const props = {
    onClick,
    disabled: disabled || isLoading,
    icon: null,
    className,
    id,
  };
  const buttonLabel = isLoading ? 'Loading...' : label;

  const component = variant === 'primary' ? (
    <Primary {...props} data-testid='PrimaryButton'>
      {icon ? <Icon src={icon} /> : null}
      {buttonLabel}
    </Primary>
  ) : (
    <Secondary {...props} data-testid='SecondaryButton'>
      {icon ? <Icon src={icon} /> : null}
      {buttonLabel}
    </Secondary>
  );

  return to ? <Link to={String(to)}>{component}</Link> : component;
};

Button.defaultProps = {
  to: '',
  icon: null,
  variant: 'primary',
  onClick: () => {},
  disabled: false,
  className: '',
  isLoading: false,
  id: '',
};
