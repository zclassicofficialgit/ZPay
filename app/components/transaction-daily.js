// @flow

import React, { Fragment } from 'react';
import styled from 'styled-components';

import { TransactionItemComponent, type Transaction } from './transaction-item';
import { TextComponent } from './text';

const Wrapper = styled.div`
  margin-top: 20px;
`;

const TransactionsWrapper = styled.div`
  border-radius: ${props => props.theme.boxBorderRadius};
  overflow: hidden;
  background-color: ${props => props.theme.colors.cardBackgroundColor};
  padding: 0;
  margin: 0;
  box-sizing: border-box;
`;

const Day = styled(TextComponent)`
  text-transform: uppercase;
  color: ${props => props.theme.colors.transactionsDate};
  font-size: ${props => `${props.theme.fontSize.regular * 0.9}em`};
  font-weight: ${props => String(props.theme.fontWeight.bold)};
  margin-bottom: 5px;
`;

type Props = {
  transactionsDate: string,
  transactions: Transaction[],
  zclPrice: number,
};

export const TransactionDailyComponent = ({ transactionsDate, transactions, zclPrice }: Props) => (
  <Wrapper data-testid='TransactionsDaily'>
    <Day value={transactionsDate} />
    <TransactionsWrapper>
      {transactions.map(
        ({
          date, type, address, amount, transactionId, confirmed, confirmations,
        }) => (
          <Fragment key={`${address}-${type}-${amount}-${date}`}>
            <TransactionItemComponent
              confirmations={confirmations}
              confirmed={confirmed}
              transactionId={transactionId}
              type={type}
              date={date}
              address={address || 'N/A'}
              amount={amount}
              zclPrice={zclPrice}
            />
          </Fragment>
        ),
      )}
    </TransactionsWrapper>
  </Wrapper>
);
