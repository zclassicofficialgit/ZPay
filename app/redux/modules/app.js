// @flow

import electronStore from '../../../config/electron-store';

import { ZCLASSIC_NETWORK, EMBEDDED_DAEMON } from '../../constants/zclassic-network';
import { NODE_SYNC_TYPES } from '../../constants/node-sync-types';

import type { Action } from '../../types/redux';

export type State = {|
  isErrorModalVisible: boolean,
  error: string | null,
  nodeSyncProgress: number,
  nodeSyncType: 'ready' | 'syncing' | 'error',
  zclassicNetwork: string,
  embeddedDaemon: boolean,
  isRefetching: boolean,
|};

// Actions
export const SHOW_ERROR_MODAL = 'SHOW_ERROR_MODAL';
export const HIDE_ERROR_MODAL = 'HIDE_ERROR_MODAL';
export const UPDATE_NODE_SYNC_STATUS = 'UPDATE_NODE_SYNC_STATUS';

export const showErrorModal = ({ error }: { error: string }) => ({
  type: SHOW_ERROR_MODAL,
  payload: {
    error,
  },
});

export const closeErrorModal = () => ({
  type: HIDE_ERROR_MODAL,
  payload: {},
});

export const updateNodeSyncStatus = ({
  nodeSyncProgress,
  nodeSyncType,
}: {
  nodeSyncProgress: number,
  nodeSyncType: $PropertyType<State, 'nodeSyncType'>,
}) => ({
  type: UPDATE_NODE_SYNC_STATUS,
  payload: {
    nodeSyncProgress,
    nodeSyncType,
  },
});

const initialState: State = {
  isErrorModalVisible: false,
  error: null,
  nodeSyncProgress: 0,
  nodeSyncType: NODE_SYNC_TYPES.SYNCING,
  zclassicNetwork: electronStore.get(ZCLASSIC_NETWORK),
  embeddedDaemon: electronStore.get(EMBEDDED_DAEMON),
  isRefetching: false,
};

// eslint-disable-next-line
export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case SHOW_ERROR_MODAL:
      return { ...state, isErrorModalVisible: true, error: action.payload.error };
    case HIDE_ERROR_MODAL:
      return { ...state, isErrorModalVisible: false, error: null };
    case UPDATE_NODE_SYNC_STATUS:
      return {
        ...state,
        nodeSyncProgress: action.payload.nodeSyncProgress,
        nodeSyncType: action.payload.nodeSyncType,
      };
    default:
      return state;
  }
};
