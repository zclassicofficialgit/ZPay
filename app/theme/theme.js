// @flow

import theme from 'styled-theming';

import { DARK, LIGHT } from '../constants/themes';
import { typography } from './typography';
import { DARK_COLORS, LIGHT_COLORS } from './colors';

// Building color object from theme keys
// All themes must have the same variables as DARK_COLORS
const colorKeys = Object.keys(DARK_COLORS);

const colors: Object = {};
colorKeys.forEach((key: string) => {
  colors[key] = theme('mode', {
    [LIGHT]: LIGHT_COLORS[key],
    [DARK]: DARK_COLORS[key],
  });
});

export const appTheme: AppTheme = {
  // General
  mode: DARK,

  // Typography
  ...typography,

  // Sizes & Spacing
  sidebarWidth: '180px',
  headerHeight: '60px',
  layoutPaddingLeft: '35px',
  layoutPaddingRight: '35px',
  layoutContentPaddingTop: '20px',
  boxBorderRadius: '3px',

  // Misc
  transitionEase: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)',

  // Colors
  colors: { ...colors },
};
