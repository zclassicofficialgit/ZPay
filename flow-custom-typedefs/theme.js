// @flow

import { ThemeSet } from 'styled-theming';

type Colors = {
  // General
  text: ThemeSet,
  success: ThemeSet,
  error: ThemeSet,
  background: ThemeSet,
  divider: ThemeSet,
  activeItem: ThemeSet,
  inactiveItem: ThemeSet,
  modalItemLabel: ThemeSet,

  // Header
  headerTitle: ThemeSet,

  // Select
  selectButtonShadow: ThemeSet,

  // Send View (Additional Options Section)
  sendAdditionalOptionsBg: ThemeSet,
  sendAdditionalOptionsBorder: ThemeSet,
  sendAdditionalInputBg: ThemeSet,
  sendAdditionalInputText: ThemeSet,

  // Dropdown
  dropdownBg: ThemeSet,
  dropdownHoveredBg: ThemeSet,
  dropdownBorder: ThemeSet,
  dropdownIconBorder: ThemeSet,
  dropdownOpenedIconBorder: ThemeSet,

  // Card
  cardBackgroundColor: ThemeSet,
  sendCardBg: ThemeSet,
  sendCardBorder: ThemeSet,

  // Buttons
  buttonPrimaryBg: ThemeSet,
  buttonPrimaryText: ThemeSet,
  buttonPrimaryDisabledBg: ThemeSet,
  buttonSecondaryBg: ThemeSet,
  buttonSecondaryText: ThemeSet,
  buttonSecondaryDisabledBg: ThemeSet,
  buttonSecondaryBorder: ThemeSet,
  buttonSecondaryHoveredBg: ThemeSet,
  buttonBorderColor: ThemeSet,

  // Transactions
  transactionSent: ThemeSet,
  transactionReceived: ThemeSet,
  transactionsDate: ThemeSet,
  transactionsItemHovered: ThemeSet,
  transactionItemBg: ThemeSet,
  transactionItemHoverBg: ThemeSet,
  transactionItemAddress: ThemeSet,
  transactionItemAddressHover: ThemeSet,
  transactionDetailsShadow: ThemeSet,
  transactionDetailsBg: ThemeSet,
  transactionDetailsRowHover: ThemeSet,
  transactionDetailsDivider: ThemeSet,
  transactionDetailsLabel: ThemeSet,
  transactionLabelText: ThemeSet,
  transactionLabelTextHovered: ThemeSet,

  // Status Pill
  statusPillBg: ThemeSet,
  statusPillLabel: ThemeSet,
  statusPillBorder: ThemeSet,

  // Sidebar
  sidebarBg: ThemeSet,
  sidebarItem: ThemeSet,
  sidebarItemActive: ThemeSet,
  sidebarActiveItemBorder:  ThemeSet,
  sidebarBorderRight: ThemeSet,
  sidebarItemHovered: ThemeSet,
  sidebarHoveredItemLabel: ThemeSet,
  sidebarLogoGradientBegin: ThemeSet,
  sidebarLogoGradientEnd: ThemeSet,
  sidebarItemHoveredBg: ThemeSet,

  // QR Code
  qrCodeWrapperBg: ThemeSet,
  qrCodeWrapperBorder: ThemeSet,

  // Forms
  inputBg: ThemeSet,
  inputBorder: ThemeSet,
  inputBorderActive: ThemeSet,

  // Wallet Summary
  walletSummaryBg: ThemeSet,
  walletSummaryBorder: ThemeSet,

  // Wallet Address
  walletAddressBg: ThemeSet,
  walletAddressBorder: ThemeSet,
  walletAddressInput: ThemeSet,
  walletAddressInputHovered: ThemeSet,
  walletAddressTooltip: ThemeSet,
  walletAddressTooltipBg: ThemeSet,

  // Console
  consoleBg: ThemeSet,
  consoleBorder: ThemeSet,

  // Settings
  settingsCardBg: ThemeSet,
  settingsLearnMore: ThemeSet,
  settingsLearnMoreHovered: ThemeSet,

  // Loading
  loadingScreenBg: ThemeSet,
  loadingScreenText: ThemeSet,
  loadingScreenProgress: ThemeSet,
};

type FontSize = {
  large: number,
  medium: number,
  regular: number,
  small: number,
};

type FontWeight = {
  bold: number,
  default: number,
  light: number,
};

type AppTheme = {
  // General
  mode: string,

  // Typography
  fontFamily: string,
  fontWeight: FontWeight,
  fontSize: FontSize,

  // Colors
  colors: Colors,

  // Spacing
  sidebarWidth: string,
  headerHeight: string,
  layoutPaddingLeft: string,
  layoutPaddingRight: string,
  layoutContentPaddingTop: string,
  boxBorderRadius: string,

  // Misc
  transitionEase: string,
};

declare type PropsWithTheme<T = {}> = {
  ...T,
  theme: AppTheme,
};
