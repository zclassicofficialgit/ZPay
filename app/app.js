// @flow

import React, { Component, Fragment } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';

import { configureStore, history } from './redux/create';
import { Router } from './router/container';
import { appTheme as theme, GlobalStyle } from './theme';
import electronStore from '../config/electron-store';
import { DARK, LIGHT, THEME_MODE } from './constants/themes';

const store = configureStore({});

type Props = {};
type State = {
  themeMode: string,
};

const getInitialTheme = () => {
  const themeInStore = String(electronStore.get(THEME_MODE));

  if (themeInStore === DARK || themeInStore === LIGHT) return themeInStore;
  return DARK;
};

export class App extends Component<Props, State> {
  state = {
    themeMode: getInitialTheme(),
  };

  componentDidMount() {
    electronStore.set(THEME_MODE, getInitialTheme());

    electronStore.onDidChange(THEME_MODE, newValue => this.setState({ themeMode: newValue }));
  }

  render() {
    const { themeMode } = this.state;

    return (
      <ThemeProvider theme={{ ...theme, mode: themeMode }}>
        <Fragment>
          <GlobalStyle />
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <Router />
            </ConnectedRouter>
          </Provider>
        </Fragment>
      </ThemeProvider>
    );
  }
}
