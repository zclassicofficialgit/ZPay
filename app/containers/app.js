// @flow

import { connect } from 'react-redux';

import { closeErrorModal } from '../redux/modules/app';
import { LayoutComponent } from '../components/layout';

import type { Dispatch } from '../types/redux';
import type { AppState } from '../types/app-state';

export type MapStateToProps = {|
  isErrorModalVisible: boolean,
  zclassicNetwork: string,
  error: string | null,
  embeddedDaemon: boolean,
|};

const mapStateToProps = ({ app }: AppState): MapStateToProps => ({
  isErrorModalVisible: app.isErrorModalVisible,
  error: app.error,
  zclassicNetwork: app.zclassicNetwork,
  embeddedDaemon: app.embeddedDaemon,
});

export type MapDispatchToProps = {|
  closeErrorModal: () => void,
|};

const mapDispatchToProps = (dispatch: Dispatch): MapDispatchToProps => ({
  closeErrorModal: () => dispatch(closeErrorModal()),
});

// $FlowFixMe
export const AppContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(LayoutComponent);
