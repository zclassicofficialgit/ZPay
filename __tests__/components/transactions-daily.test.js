// @flow

import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import dateFns from 'date-fns';
import 'jest-dom/extend-expect';

import { TransactionDailyComponent } from '../../app/components/transaction-daily';
import { appTheme } from '../../app/theme';

let originalDate;
const fixedDate = new Date('2018-02-28T09:39:59');

beforeAll(() => {
  // $FlowFixMe
  dateFns.format = jest.fn(() => '17:01 PM');

  originalDate = global.Date;
  global.Date = class extends Date {
    constructor() {
      super();

      return fixedDate;
    }
  };
});
afterAll(() => {
  global.Date = originalDate;
  // $FlowFixMe
  dateFns.format.mockRestore();
  cleanup();
});

describe('<TransactionDailyComponent />', () => {
  describe('render()', () => {
    test('should render user daily transactions', () => {
      const { container } = render(
        <ThemeProvider theme={appTheme}>
          <TransactionDailyComponent
            transactionsDate='2019-02-20T19:31:57.117Z'
            zclPrice={1.345}
            transactions={[
              {
                type: 'receive',
                transactionId: 's0a8das098fgh2348a',
                address: '123456789123456789123456789123456789',
                amount: 1.7891,
                zclPrice: 1.345,
                date: '2019-02-20T19:31:57.117Z',
                theme: appTheme,
                fees: 0.0001,
                confirmations: 10,
                confirmed: true,
              },
              {
                type: 'send',
                transactionId: '0asd908fgj90f01',
                address: '123456789123456789123456789123456789',
                amount: 0.8458,
                zclPrice: 1.344,
                date: '2019-02-20T19:31:57.117Z',
                theme: appTheme,
                fees: 0.0001,
                confirmed: false,
                confirmations: 3,
              },
            ]}
          />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
