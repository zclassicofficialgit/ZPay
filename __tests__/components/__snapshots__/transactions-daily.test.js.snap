// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionDailyComponent /> render() should render user daily transactions 1`] = `
<div>
  <div
    class="sc-jKJlTe gjnbgQ"
    data-testid="TransactionsDaily"
  >
    <p
      class="sc-hMqMXs gnhPXm sc-htpNat pRhzD"
    >
      2019-02-20T19:31:57.117Z
    </p>
    <div
      class="sc-eNQAEJ dtrSYf"
    >
      <div
        class="sc-jzJRlG bgSReS sc-bdVaJa bWRfen"
        id="transaction-item-s0a8das098fgh2348a"
      >
        <div
          class="sc-bdVaJa eFCRqo"
        >
          <div
            class="sc-ckVGcZ gzxdCf sc-bdVaJa eFCRqo"
          >
            <img
              alt="Transaction Type Icon"
              class="sc-cSHVUG fQARKX"
              src="[object Object]"
            />
            <div
              class="sc-kGXeez iHFhdz sc-bwzfXH iiePXZ"
              width=""
            >
              <p
                class="sc-kAzzGY edxhBx sc-htpNat ccIOzB"
              >
                receive
              </p>
              <p
                class="sc-kgoBCf goviHQ sc-htpNat pRhzD"
              >
                17:01 PM
              </p>
            </div>
          </div>
          <p
            class="sc-chPdSV dTXjho sc-htpNat pRhzD"
          >
            123456789123456789123456789123456789
          </p>
        </div>
        <div
          class="sc-bwzfXH jExuEg"
          width=""
        >
          <p
            class="sc-htpNat dztgbW"
          >
            +TAZ 1.789
          </p>
          <p
            class="sc-kgoBCf goviHQ sc-htpNat pRhzD"
          >
            +USD $2.406
          </p>
        </div>
      </div>
      <div
        class="sc-jzJRlG bgSReS sc-bdVaJa bWRfen"
        id="transaction-item-0asd908fgj90f01"
      >
        <div
          class="sc-bdVaJa eFCRqo"
        >
          <div
            class="sc-ckVGcZ gzxdCf sc-bdVaJa eFCRqo"
          >
            <img
              alt="Transaction Type Icon"
              class="sc-cSHVUG fQARKX"
              src="[object Object]"
            />
            <div
              class="sc-kGXeez iHFhdz sc-bwzfXH iiePXZ"
              width=""
            >
              <p
                class="sc-kAzzGY gABhMI sc-htpNat ccIOzB"
              >
                send
              </p>
              <p
                class="sc-kgoBCf goviHQ sc-htpNat pRhzD"
              >
                17:01 PM
              </p>
            </div>
          </div>
          <p
            class="sc-chPdSV dTXjho sc-htpNat pRhzD"
          >
            123456789123456789123456789123456789
          </p>
        </div>
        <div
          class="sc-bwzfXH jExuEg"
          width=""
        >
          <p
            class="sc-htpNat jAbxuE"
          >
            -TAZ 0.846
          </p>
          <p
            class="sc-kgoBCf goviHQ sc-htpNat pRhzD"
          >
            -USD $1.138
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
